src/mito2/src/config.rs

```diff
@@ -48,6 +49,30 @@ const PAGE_CACHE_SIZE_FACTOR: u64 = 8;
 /// Use `1/INDEX_CREATE_MEM_THRESHOLD_FACTOR` of OS memory size as mem threshold for creating index
 const INDEX_CREATE_MEM_THRESHOLD_FACTOR: u64 = 16;
 
+define_mem_size_enum!(
+    /// Global write buffer size threshold to trigger flush.
+    GlobalWriteBufferSize, GLOBAL_WRITE_BUFFER_SIZE_FACTOR, ReadableSize::gb(1), Some(ReadableSize::gb(1)));
```

Comment:
Do we need to support unlimited size for all options? Also, if we decouple the example config from the default value, we can use the `auto` value by not specifying this option in the TOML. Then we can still use `ReadableSize`.

Also, there is an issue with the mem size enum: We can't precompute the size while the enum is `Auto`. We need to compute the size each time we call `as_bytes()`.

We may also have some options that we want to disable by default and can't get the default value automatically.

Maybe a better way is to support our requirements in https://github.com/zyy17/toml2docs

Reply:
I thought we could remove the `unlimited`, what do you think? @zhongzc .

> We can't precompute the size while the enum is Auto. We need to compute the size each time we call as_bytes().

Yes, but I think that is good. We don't call it many times.

Reply:
We need to define a new type for each size just because we need a different default value. At least we can define a function for each default value and provide a method for `get_or_default()`.
```rust
enum AutoReadableSize {}

let size = AutoReadableSize::Auto;
let x = size.get_or_default(default_global_write_buffer_size);
```